expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/trait/operation/OperationResponseTrait.kt

```diff
@@ -0,0 +1,34 @@
+package com.expediagroup.sdk.rest.trait.operation
+
+import com.fasterxml.jackson.core.type.TypeReference
+
+/**
+ * Marker interface for operations that have a response.
+ */
+interface OperationResponseTrait : OperationTrait
+
+/**
+ * Marker interface for operations that do not have a response body.
+ */
+interface OperationNoResponseBodyTrait : OperationResponseTrait
```

Comment:
Why do we need this trait? Can an operation just not implement the response body trait?

Reply:
We need this for `Client` to be able to infer the type of responses and return types, and to know which execute method to use. Based on our plan, the `Client` will look somewhat like this:

```kotlin
class Client {
  fun execute(operation: OperationNoResponseBodyTrait): Response
  fun <T> execute(operation: OperationResponseBodyTrait<T>): Response<T>
}
```

So by having the `OperationNoResponseBodyTrait`, we can make use of the type system by making it automatically know which execution method to use and when a return type is expected. Replacing the `OperationNoResponseBodyTrait` type in the execute method with one of its parents can make things complicated so that's not an option too.